### PR TITLE
Bluetooth: audio: Factor out PACS related API to internal header

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This moves the PACS API to internal header, as it duplicates the
bt_capabilities API.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

**Comment by Thalley** [comment]
Some of the changes here were also planned for https://github.com/zephyrproject-rtos/zephyr/issues/45622 but it is fine to fix them here :)

File: include/zephyr/bluetooth/audio/stream.h
Comment (Thalley) [inline include/zephyr/bluetooth/audio/stream.h:3]: For CAP we need to add another layer on top of the `bt_audio_stream`, essentially making the `bt_audio_stream` into a `bt_bap_stream`, as the relationsship between BAP streams and CAP streams are similar to the relationsship between BAP and ISO, i.e. it is essentially an extension of the other. 

I had planned to rename `struct bt_audio_stream` into `struct bt_bap_stream` and move it into a `bap.h`. 

This can still be done after this change though, but it kind of defeats the purpose of this move. 

The structure I had in mind was a `bap.h` (containing BAP specific resources) and a `cap.h` containing CAP specific resources, and then keep an `audio.h` header for things that span across LE Audio in general (like the assigned numbers).

**Review by Thalley — CHANGES_REQUESTED** [review]
The changes suggested by this PR is a significant improvement over the current, but I fear that it may be blocking some future work in the way that it is structured. 

Perhaps we need to figure out the header structure for LE Audio, BAP and CAP before we start moving things around.

File: include/zephyr/bluetooth/audio/audio.h
Comment (MariuszSkamra) [reply include/zephyr/bluetooth/audio/audio.h:1363]: Ah, you have right. I recall we had a discussion, and as you said, we were agreed to remove callbacks from capabilities. 
I've realized it's used by broadcast sink as well when I saw the failing CI on this PR. Let me update the PR.

File: include/zephyr/bluetooth/audio/codec.h
Comment (MariuszSkamra) [reply include/zephyr/bluetooth/audio/codec.h:3]: I thought it would be LE Audio specific as it defines the codec configurations defined in BAP.

File: include/zephyr/bluetooth/audio/codec.h
Comment (MariuszSkamra) [reply include/zephyr/bluetooth/audio/codec.h:1072]: You have right, I saw that, but I did not want to make any further changes (changing the naming) since I did not know this change will be accepted. If we agree the code can be moved to this file, I'll add another patch on top to refactor the naming.

File: include/zephyr/bluetooth/audio/codec.h
Comment (Thalley) [reply include/zephyr/bluetooth/audio/codec.h:3]: Indeed. Well, I guess LE Audio specific, as we will likely reuse them in CAP as well.

**Comment by Thalley** [comment]
@MariuszSkamra Is your plan to continue with this? Otherwise I'm planning on starting https://github.com/zephyrproject-rtos/zephyr/issues/45622 soon, and it would be foolish to have two active PRs in progress for this. If you are planning on continuing with this soon, then I'll put https://github.com/zephyrproject-rtos/zephyr/issues/45622 on hold until this has been merged.

**Review by Thalley — APPROVED** [review]
The fact that this does not require any changes in any samples, is a good indication that this is a good change :)

**Review by sjanc — APPROVED** [review]

**Pull request merged**
